File: app/src/main/java/com/playgilround/schedule/client/calendar/listener/OnScheduleScrollListener.java
package com.playgilround.schedule.client.calendar.listener;

import android.util.Log;
import android.view.GestureDetector;
import android.view.MotionEvent;

import com.playgilround.schedule.client.calendar.ScheduleLayout;

/**
 * 달력, 스케줄 부분 스크롤 리스너.
 */
public class OnScheduleScrollListener extends GestureDetector.SimpleOnGestureListener {

    private ScheduleLayout mScheduleLayout;
    private static final String TAG = OnScheduleScrollListener.class.getSimpleName();

    public OnScheduleScrollListener(ScheduleLayout scheduleLayout) {
        mScheduleLayout = scheduleLayout;
    }

    @Override
    public boolean onDown(MotionEvent e) {
        return true;
    }

    @Override
    public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
        Log.d(TAG, "onScroll...");
        mScheduleLayout.onCalendarScroll(distanceY);
        return true;
    }
}

File: app/src/main/java/com/playgilround/schedule/client/calendar/listener/OnScheduleScrollListener.kt
package com.playgilround.schedule.client.calendar.listener

import android.view.GestureDetector
import android.view.MotionEvent
import com.playgilround.schedule.client.calendar.ScheduleLayout

class OnScheduleScrollListener constructor(private val mScheduleLayout: ScheduleLayout): GestureDetector.SimpleOnGestureListener() {

    override fun onDown(e: MotionEvent): Boolean {
        return true
    }

    override fun onScroll(e1: MotionEvent?, e2: MotionEvent?, distanceX: Float, distanceY: Float): Boolean {
        mScheduleLayout.onCalendarScroll(distanceY)
        return true
    }
}